internal/tui/views/trafficwatch/left_pane.go

```diff
@@ -284,23 +281,25 @@ func (l *LeftPane) renderRequestItem(req *filemanager.RequestMetadata, selected
 	}
 
 	// Format timestamp (strip milliseconds and timezone)
-	timestamp := req.DoneAt
-	formattedTime := timestamp.Format("2006-01-02 15:04:05")
+	formattedTime := req.DoneAt.Format("2006-01-02 15:04:05")
 
 	// Properly render protocol
-	protocol := strings.ToUpper(string(req.Protocol))
-	if strings.Contains(protocol, "GRPC") {
+	var protocol string
+	switch req.Protocol {
+	case filemanager.ProtocolGRPC:
 		protocol = "gRPC"
-	} else {
-		protocol = "HTTP"
+	default:
+		protocol = strings.ToUpper(string(req.Protocol))
 	}
 
-	line := lipgloss.NewStyle().Width(l.width).Render(
-		fmt.Sprintf("%s  %-5s  %-6s  %s%s", formattedTime, protocol, method, host, fullPath),
-	)
-
-	content := fmt.Sprintf("%s", line)
+	// date-time  protocol  host
+	line1 := fmt.Sprintf("%s  %-5s  ", formattedTime, protocol)
+	line1 += truncateURL(host, l.width-len(line1)-1)
+	// method  fullPath
+	line2 := fmt.Sprintf("%-6s  ", method)
+	line2 += truncateURL(fullPath, l.width-len(line2)-1)
```

Comment:
It's better to use the lipglos.Width(line2) cause it take in account other things i think